### ISSUE TITLE
label parties on bar chart

### ISSUE THREAD

**Issue by aspiers** [swapmyvote/swapmyvote#825]
For me a large part of the struggle with the bar chart is that the parties aren't labelled (only visible via mouse hover), and I haven't perfectly internalised the various party colours.  It should be pretty easy to fix this, e.g. via one or both of:

- Include the party logo in the bar.
- Include the party name in the bar (rotated vertically to fit, using CSS).

_Originally posted by @aspiers in https://github.com/swapmyvote/swapmyvote/issues/816#issuecomment-2179206246_

**Comment by baob** [comment]
@stevebaxter could you pick this one up maybe ?

**Comment by stevebaxter** [comment]
Looking now, finally back in phone range. I'm not getting the colours in the bar chart, should that be there now? I think doing this depends on the capabilities of the graphing package we're using.

**Comment by aspiers** [comment]
I thought @baob previously encountered and fixed this colours bug - I could be wrong, but worth checking that you have the latest code and the webpacker build is up to date also.

**Comment by stevebaxter** [comment]
Ah I needed to reseed the DB, getting the colours now.

**Comment by aspiers** [comment]
It would be amazing to get little party logos on the chart too but one step at a time :-)

**Comment by stevebaxter** [comment]
Yeah, that's what I'm looking at. It's not easy though, and do we have a permitted source for the logos?

**Comment by aspiers** [comment]
Not sure about a permitted source but I'm sure @tdg or someone else in the community can advise.  I suggest we start with just the text labels rotated vertically, and then worry about logos after that?

**Comment by stevebaxter** [comment]
There's not enough space to put the labels in the bars because some of the bars are very short. We can put the labels at the bottom, I think we can use shortened versions of the names. Something like this (I haven't shortened the names yet):
<img width="602" alt="Screenshot 2024-06-23 at 17 59 18" src="https://github.com/swapmyvote/swapmyvote/assets/5228091/f37f584a-e3d6-4b1f-a1e7-bf25fbf980a1">

**Comment by aspiers** [comment]
This looks great to me!! :clap:

**Comment by aspiers** [comment]
We could also use specific abbreviations in this case, e.g. "Lib Dem" is nice and short.  Not sure how to shorten "Conservatives" other than "Tories" which are perhaps inappropriate

**Comment by stevebaxter** [comment]
Yep, trying to add the short names to the `Party` object in the DB now. Going for Lab, Con, LD, Plaid, SNP, Ref.

**Comment by baob** [comment]
> trying to add the short names to the `Party` object in the DB

Wouldn't the smv_code to the job ?

Please let me know what you're thinking here, I really think adding another field to the party table might be overkill, there's probably a lookup in the Party model that already does the job. A video/audio call welcome.

**Comment by aspiers** [comment]
If I understand right, `smv_code` was never supposed to be used as a human-readable display name - isn't it just intended for internal use?

We also have `short_name` in the `Recommendation` model, which _does_ seem like the right thing, but is in the wrong model :frowning_face:  Would be great to deduplicate all this and have the following all in `Party`:

- machine-readable name only for internal use (`smv_code`)
- full length human-readable display name (currently `name`)
- short version of human-readable display name (currently `Recommendation#short_name` but should be `Party#short_name`)

**Comment by baob** [comment]
You're right @aspiers I made the comment without first looking at the sprawling mass of code that has loads of different ways of identifying parties.

Recommendation#short_name was originally made for the recommendations table we no longer render. It might be a good time to move it to the Party model.

Or copy it in to the Party model.

If it was inserted into the REFERENCE_DATA hash, that's keyed by smv_code ... which should be a handy way to get at that data ?

**Comment by stevebaxter** [comment]
That's what I've done. I think! OK, it looks like this, what do you think?
<img width="620" alt="Screenshot 2024-06-23 at 18 39 06" src="https://github.com/swapmyvote/swapmyvote/assets/5228091/398d56b9-5187-4597-b8d5-e8839d711ca0">

**Comment by baob** [comment]
Reads fine on the page. And if you've done it by adding attributes to the hashes in REFERENCE_DATA that sounds great.

**Comment by stevebaxter** [comment]
Yep. Just coming into Portsmouth, will try and raise a PR quickly so you guys can look at it while I drive home this evening.

**Comment by aspiers** [comment]
Awesome!  It's a pleasure to work with you two :)
